easybuild/easyblocks/t/tinker.py

```diff
@@ -0,0 +1,154 @@
+
+        comp_dirs = {
+            toolchain.INTELCOMP: 'intel',
+            toolchain.GCC: 'gfortran',
+        }
+        comp_fam = self.toolchain.comp_family()
+        comp_dir = comp_dirs.get(comp_fam)
+        if comp_dir is None:
+            self.log.error("Failed to determine compiler directory for %s (known: %s)" % (comp_fam, comp_dirs))
+
+        self.build_subdir = os.path.join(os_dir, comp_dir)
+        self.log.info("Using build scripts from %s subdirectory" % self.build_subdir)
+
+        # patch 'link.make' script to use FFTW provided via EasyBuild
+        link_make_fp = os.path.join(self.cfg['start_dir'], self.build_subdir, 'link.make')
+        for line in fileinput.input(link_make_fp, inplace=1, backup='.orig'):
+            line = re.sub(r"libfftw3_threads.a libfftw3.a", r"-L$EBROOTFFTW/lib -lfftw3_threads -lfftw3", line)
+            sys.stdout.write(line)
+
+    def build_step(self):
+        """Custom build procedure for TINKER."""
+        source_dir = os.path.join(self.cfg['start_dir'], 'source')
+        try:
+            os.chdir(source_dir)
+        except OSError, err:
+            self.log.error("Failed to move to %s: %s" % (source_dir, err))
+
+        run_cmd(os.path.join(self.cfg['start_dir'], self.build_subdir, 'compile.make'))
+        run_cmd(os.path.join(self.cfg['start_dir'], self.build_subdir, 'library.make'))
+        run_cmd(os.path.join(self.cfg['start_dir'], self.build_subdir, 'link.make'))
+
+    def test_step(self):
+        """Custom built-in test procedure for TINKER."""
+        if self.cfg['runtest']:
+            # copy tests, params and built binaries to temporary directory for testing
+            tmpdir = tempfile.mkdtemp()
+            testdir = os.path.join(tmpdir, 'test')
+
+            mkdir(os.path.join(tmpdir, 'bin'))
+            binaries = glob.glob(os.path.join(self.cfg['start_dir'], 'source', '*.x'))
+            try:
+                for binary in binaries:
+                    shutil.copy2(binary, os.path.join(tmpdir, 'bin', os.path.basename(binary)[:-2]))
+                shutil.copytree(os.path.join(self.cfg['start_dir'], 'test'), testdir)
+                shutil.copytree(os.path.join(self.cfg['start_dir'], 'params'), os.path.join(tmpdir, 'params'))
+            except OSError, err:
+                self.log.error("Failed to copy binaries and tests to %s: %s" % (tmpdir, err))
+
+            try:
+                os.chdir(testdir)
+            except OSError, err:
+                self.log.error("Failed to move to %s to run tests: %s" % (testdir, err))
+
+            # run all tests via the provided 'run' scripts
+            tests = glob.glob(os.path.join(testdir, '*.run'))
+            # gpcr takes too logn (~1h), ifabp fails due to input issues (?)
+            tests = [t for t in tests if not (t.endswith('gpcr.run') or t.endswith('ifabp.run'))]
+            for test in tests:
+                run_cmd(test)
```

Comment:
doesn't really matter, does it?

it checks the exit code, and nicely logs the output, so the defaults are fine